In [112]:
import pandas as pd

from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import accuracy_score, precision_score, recall_score

# Cargar el archivo CSV
df1 = pd.read_csv('anexo02.csv', encoding='ISO-8859-1')


# Examinar los primeros registros del archivo
df1


Lugar  Especie  Tamaño Peso  VidaPromedio  Camada  CicloRepr  \
0      Cantabria   Cabra   176.0   28            18     3.0        155   
1        Badajoz   Cabra   173.0  100            13     2.0        149   
2    Guadalajara   Cabra   167.0  124            13     3.0        152   
3       Cataluña   Cabra   163.0   43            19     1.0        149   
4        Almería   Cabra   162.0   80            13     3.0        150   
..           ...     ...     ...  ...           ...     ...        ...   
585        Ávila  Bufalo     2.6  826            22     1.0        327   
586       Zamora  Bufalo     2.7  313            15     1.0        328   
587    Tarragona  Bufalo     2.4  725            20     1.0        328   
588     Alicante  Bufalo     2.5  346            17     1.0        330   
589     Albacete  Bufalo     2.4  455            16     1.0        331   

     Calidad de la carne  
0                      1  
1                      1  
2                      1  
3                      3  
4                      1  
..                   ...  
585                    2  
586                    4  
587                    2  
588                    2  
589                    4  

[590 rows x 8 columns]

In [113]:
# Verificar si la columna "Calidad de la carne" existe
if "Calidad de la carne" in df1:
    print(df1["Calidad de la carne"].describe())
else:
    print("La columna 'Calidad de la carne' no se encuentra en el archivo.")


count    590.000000
mean       2.962712
std        1.406497
min        1.000000
25%        2.000000
50%        3.000000
75%        4.000000
max        5.000000
Name: Calidad de la carne, dtype: float64


In [114]:
from sklearn.model_selection import train_test_split

# Eliminar filas con valores faltantes (si es necesario)
df1 = df1.dropna()

# Definir variables predictoras (X) y la variable objetivo (y)
X = df1.drop(columns=["Calidad de la carne"])
y = df1["Calidad de la carne"]

# Dividir el conjunto de datos en entrenamiento (70%) y prueba (30%)
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42)


In [115]:
df1

Lugar  Especie  Tamaño Peso  VidaPromedio  Camada  CicloRepr  \
0      Cantabria   Cabra   176.0   28            18     3.0        155   
1        Badajoz   Cabra   173.0  100            13     2.0        149   
2    Guadalajara   Cabra   167.0  124            13     3.0        152   
3       Cataluña   Cabra   163.0   43            19     1.0        149   
4        Almería   Cabra   162.0   80            13     3.0        150   
..           ...     ...     ...  ...           ...     ...        ...   
585        Ávila  Bufalo     2.6  826            22     1.0        327   
586       Zamora  Bufalo     2.7  313            15     1.0        328   
587    Tarragona  Bufalo     2.4  725            20     1.0        328   
588     Alicante  Bufalo     2.5  346            17     1.0        330   
589     Albacete  Bufalo     2.4  455            16     1.0        331   

     Calidad de la carne  
0                      1  
1                      1  
2                      1  
3                      3  
4                      1  
..                   ...  
585                    2  
586                    4  
587                    2  
588                    2  
589                    4  

[590 rows x 8 columns]

In [116]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import accuracy_score, precision_score, recall_score

# Cargar el archivo CSV
df1 = pd.read_csv('anexo02.csv', encoding='ISO-8859-1')


# Eliminar filas con valores faltantes (si es necesario)
df1 = df1.dropna()

# Definir variables predictoras (X) y la variable objetivo (y)
X = df1.drop(columns=["Calidad de la carne"])
y = df1["Calidad de la carne"]

# Identificar columnas categóricas
categorical_columns = X.select_dtypes(include=['object']).columns

# Convertir columnas categóricas en variables numéricas usando one-hot encoding
X = pd.get_dummies(X, columns=categorical_columns)

# Dividir el conjunto de datos en entrenamiento y prueba
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42)

# Entrenar el modelo de regresión lineal
lr_model = LinearRegression()
lr_model.fit(X_train, y_train)

# Predecir con el modelo entrenado
y_pred_lr = lr_model.predict(X_test)

# Evaluar el modelo
print("Precisión de la Regresión Lineal:", lr_model.score(X_test, y_test))


Precisión de la Regresión Lineal: -571165618067095.2


In [117]:
from sklearn.linear_model import LinearRegression
from sklearn.metrics import accuracy_score, precision_score, recall_score

# Entrenar modelo de regresión lineal
lr_model = LinearRegression()
lr_model.fit(X_train, y_train)

# Predecir con el modelo entrenado
y_pred_lr = lr_model.predict(X_test)

# Evaluación del modelo
print("Precisión de la Regresión Lineal:", lr_model.score(X_test, y_test))


Precisión de la Regresión Lineal: -571165618067095.2


In [118]:
from sklearn.tree import DecisionTreeClassifier

# Entrenar modelo de árbol de decisión
dt_model = DecisionTreeClassifier()
dt_model.fit(X_train, y_train)

# Predecir con el modelo entrenado
y_pred_dt = dt_model.predict(X_test)

# Evaluación del modelo
print("Precisión del Árbol de Decisión:", dt_model.score(X_test, y_test))


Precisión del Árbol de Decisión: 0.2598870056497175


In [119]:
from sklearn.neighbors import KNeighborsClassifier

# Entrenar modelo K-Nearest Neighbors
knn_model = KNeighborsClassifier()
knn_model.fit(X_train, y_train)

# Predecir con el modelo entrenado
y_pred_knn = knn_model.predict(X_test)

# Evaluación del modelo
print("Precisión del K-Nearest Neighbors:", knn_model.score(X_test, y_test))


Precisión del K-Nearest Neighbors: 0.23163841807909605


In [120]:
from sklearn.metrics import mean_squared_error, mean_absolute_error, r2_score

# Evaluar el rendimiento del modelo de regresión lineal
mse = mean_squared_error(y_test, y_pred_lr)
mae = mean_absolute_error(y_test, y_pred_lr)
r2 = r2_score(y_test, y_pred_lr)

print("Evaluación de la Regresión Lineal")
print("Error Cuadrático Medio (MSE):", mse)
print("Error Absoluto Medio (MAE):", mae)
print("R² Score:", r2)


Evaluación de la Regresión Lineal
Error Cuadrático Medio (MSE): 1093909138985026.9
Error Absoluto Medio (MAE): 19049552.36277987
R² Score: -571165618067095.2


Interpretación general:

Error Cuadrático Medio (MSE)
El MSE (Mean Squared Error) es una medida común utilizada para evaluar la precisión de un modelo de regresión. En este caso tenemos:

MSE: 1093909138985026.9

Este valor es extremadamente alto y sugiere que el modelo está teniendo problemas significativos en predecir los valores de salida. Un MSE típico para un problema bien modelado suele ser mucho menor, por ejemplo, entre 0.01 y 10.000.

Error Absoluto Medio (MAE)
El MAE (Mean Absolute Error) también es una métrica importante para evaluar modelos de regresión. Aquí tenemos:

MAE: 19049552.36277987

Similar al MSE, este valor es muy alto y indica que el modelo está cometiendo errores grandes en sus predicciones.

R² Score
El R² score (coefficient of determination) mide la proporción del varianza total explicada por el modelo. Tenemos:

R² Score: -571165618067095.2

Este valor negativo es anormal y sugiere que el modelo no está explicando nada de la varianza total. De hecho, parece que el modelo podría estar peor que un modelo que siempre predice el mismo valor constante.